### PR TITLE
Stylus cargo stylus

### PR DESCRIPTION
**Pull request by anegg0**

### CODE REVIEW

**Comment by srinjoyc** [comment]
Content is looking good but wondering if it would make sense to aggregate the tables. IE. Cargo Stylus commands reference could maybe be combined with Examples usage by adding another column?

**Comment by srinjoyc** [comment]
For the pre-req's, should we use the same component as the quickstart?

File: arbitrum-docs/stylus/using-stylus-cli.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+| [Verify contracts](https://docs.arbitrum.io/stylus/how-tos/verify-contracts)                       | Ensure that your Stylus contracts are correctly verified. Step-by-step instructions on how to verify your contracts using `cargo-stylus`.   |
+|[Run a Stylus dev node](https://docs.arbitrum.io/run-arbitrum-node/run-local-dev-node)             | Learn how to run a local Arbitrum dev node to test your Stylus contracts.                                                                    |
+
+#### Additional resources
+
+#### [Troubleshooting](https://docs.arbitrum.io/stylus/troubleshooting-building-stylus): solve the most common issues.
+
+#### [cargo-stylus repository](https://github.com/OffchainLabs/stylus): consult cargo stylus' source code. 
+
+This overview page serves as the starting point for mastering the CLI tools available for Stylus development. From optimizing your contracts to debugging and verifying them, the `cargo-stylus` toolset is integral to a smooth development experience.
```
Comment (srinjoyc) [inline arbitrum-docs/stylus/using-stylus-cli.mdx:106]:
```suggestion
```

I think this would go on top / consolidate with the intro section of this page.

**Comment by srinjoyc** [comment]
Some formatting issues here:
image

**Comment by srinjoyc** [comment]
Should VS Code be a pre-requisite for the CLI?

**Comment by srinjoyc** [comment]
We might also want to update the top to provide a quick functionality overview. IE. "The CLI provides some essential functionality for your development workflow including deploying, verifying and caching your contracts amongst other helpful tools like debugging and transaction replay.

File: arbitrum-docs/stylus/using-stylus-cli.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+description: 'Get started with Stylus CLI, a Rust toolkit for developing Stylus contracts'
+author: 'anegg0'
+sme: 'anegg0'
+sidebar_position: 2
+target_audience: Developers writing Stylus contracts in Rust using Stylus
+---
+
+This guide will get you started using [cargo stylus](https://github.com/OffchainLabs/cargo-stylus), a CLI toolkit to help developers manage, compile, and optimize their Stylus contracts efficiently.
+
+This overview is meant to help discover cargo stylus' tools and learn to use them effectively.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/using-stylus-cli.mdx:13]:
```suggestion
This overview will help you discover and learn how to uses cargo stylus tools.
```

File: arbitrum-docs/stylus/using-stylus-cli.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+### Installing cargo stylus
+
+Cargo stylus is available as a plugin to the standard cargo tool used for developing Rust programs.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/using-stylus-cli.mdx:17]:
```suggestion
Cargo stylus is a plugin to the standard cargo tool for developing Rust programs.
```

File: arbitrum-docs/stylus/using-stylus-cli.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Prerequisites
+
+<details>
+<summary>Rust toolchain</summary>
+
+Follow the instructions on [Rust Lang's installation page](https://www.rust-lang.org/tools/install) to install a complete Rust toolchain (v1.81 or older, v1.82 is currently not supported) on your system. After installation, ensure you can access the programs `rustup`, `rustc`, and `cargo` from your preferred terminal application.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/using-stylus-cli.mdx:24]:
```suggestion
Follow the instructions on [Rust Lang's installation page](https://www.rust-lang.org/tools/install) to install a complete Rust toolchain (v1.81 or older; v1.82 is currently not supported) on your system. After installation, ensure you can access the programs `rustup`, `rustc`, and `cargo` from your preferred terminal application.
```

File: arbitrum-docs/stylus/using-stylus-cli.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+- [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens): Immediately highlights errors and warnings in your code
+- [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml): Improves syntax highlighting and other features for TOML files, often used in Rust projects
+- [Dependi](https://marketplace.visualstudio.com/items?itemName=fill-labs.dependi): Helps manage Rust crate versions directly from the editor
+
+ </details>
+
+<details>
+<summary>Docker</summary>
+
+The testnode we will use as well as some `cargo stylus` commands require Docker to operate.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/using-stylus-cli.mdx:45]:
```suggestion
We will use the testnet, and some `cargo stylus` commands will require Docker to operate.
```

File: arbitrum-docs/stylus/using-stylus-cli.mdx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+<summary>Foundry's Cast</summary>
+
+[Foundry's Cast](https://book.getfoundry.sh/cast/) is a command-line tool that allows you to interact with your EVM contracts.
+
+</details>
+
+<details>
+<summary>Nitro devnode</summary>
+
+Stylus is available on Arbitrum Sepolia, but we'll use nitro devnode which has a pre-funded wallet saving us the effort of wallet provisioning or running out of tokens to send transactions.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/using-stylus-cli.mdx:61]:
```suggestion
Stylus is available on Arbitrum Sepolia, but we'll use nitro devnode, which has a pre-funded wallet, saving us the effort of wallet provisioning or running out of tokens to send transactions.
```

File: arbitrum-docs/stylus/using-stylus-cli.mdx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+Add WASM ([WebAssembly](https://webassembly.org/)) as a build target for the specific Rust toolchain you are using. The below example sets your default Rust toolchain to 1.80 as well as adding the WASM build target:
+
+```shell
+rustup default 1.80
+rustup target add wasm32-unknown-unknown --toolchain 1.80
+```
+
+You can verify that cargo stylus is installed by running `cargo stylus -V` in your terminal, returning something like:`stylus 0.5.6`
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/using-stylus-cli.mdx:89]:
```suggestion
You can verify the cargo stylus installation by running `cargo stylus -V` in your terminal, returning something like:`stylus 0.5.6`
```

**Review by pete-vielhaber — APPROVED** [review]
- Minor suggestions for easier reading

**Pull request merged**